base/settings/R/get.site.info.R

```diff
@@ -0,0 +1,205 @@
+#' Extract and validate site information from settings or CSV file
+#'
+#' @param settings PEcAn settings list containing site information (optional)
+#' @param csv_path Path to a CSV file containing site information (optional)
+#' @param strict_checking Logical. If TRUE, will validate coordinates more strictly
+#'
+#' @return A data frame with site_id, site_name, lat, lon, and str_id
+#' @export get.site.info
+#'
+#' @details This function extracts and validates site information from either a PEcAn settings 
+#'          object or a CSV file. At least one input must be provided. If both are provided,
+#'          the settings object takes precedence.
+#'          
+#'          If using a CSV file, it must contain at minimum the columns: site_id, lat, and lon.
+#'          The column site_name is optional and will default to site_id if not provided.
+#'
+#' @examples
+#' \dontrun{
+#' # From settings object
+#' settings <- PEcAn.settings::read.settings("pecan.xml")
+#' site_info <- PEcAn.settings::get.site.info(settings)
+#'
+#' # From CSV file
+#' site_info <- PEcAn.settings::get.site.info(csv_path = "sites.csv")
+#' }
+get.site.info <- function(settings = NULL, csv_path = NULL, strict_checking = TRUE) {
```

Comment:
In general, I found this function is too complicated. In my previous code, I usually just use a single chunck of code to do the conversion, which I think might be useful in your case: 
```suggestion
site_info <- settings$run %>% 
                                 purrr::map('site')%>% 
                                 purrr::map(function(site.list){
                                   #conversion from string to number
                                   site.list$lat <- as.numeric(site.list$lat)
                                   site.list$lon <- as.numeric(site.list$lon)
                                   list(site.id=site.list$id, lat=site.list$lat, lon=site.list$lon, site_name=site.list$name)
                                 }) %>% 
                                 dplyr::bind_rows() %>% 
                                 as.list())
```